### ISSUE TITLE
Can't build with nuxt 3.12.1, missing 

### ISSUE THREAD

**Issue by ManUtopiK** [nuxt/nuxt#27532]
### Environment

- Operating System: `Linux`
- Node Version:     `v20.11.1`
- Nuxt Version:     `3.12.1`
- CLI Version:      `3.12.0`
- Nitro Version:    `2.9.6`
- Package Manager:  `pnpm@9.1.4`
- Builder:          `-`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

I've just upgraded to nuxt 3.12.1 with `npx nuxi@latest upgrade --force` and now I got this error when building with `pnpm build`.

### Describe the bug

Output :
![image](https://github.com/nuxt/nuxt/assets/188172/c642e953-ef39-4b82-bdfd-7b9fd160760f)


### Additional context

I use nuxt in a pnpm monorepo.

### Logs

<details>
  <summary>Log</summary>

```bash
[14:25:12]  ERROR  Nuxt Build Error: virtual:nuxt:/home/manu/Sites/unpress/packages/cms/.nuxt/imports.mjs (1:260): "useScriptGoogleAdsense" is not exported by "../../node_modules/.pnpm/nuxt@3.12.1_@opentelemetry+api@1.9.0_@parcel+watcher@2.4.1_@types+node@20.14.2_@unocss+reset@_yxjhw5zvsqxaxtrtsod25wbmiu/node_modules/nuxt/dist/app/composables/script-stubs.js", imported by "virtual:nuxt:/home/manu/Sites/unpress/packages/cms/.nuxt/imports.mjs".
file: virtual:nuxt:/home/manu/Sites/unpress/packages/cms/.nuxt/imports.mjs:1:260

1: export { useConsentScriptTrigger, useAnalyticsPageEvent, useElementScriptTrigger, useScript, useScriptGoogleAnalytics, useScriptPlausibleAnalytics, useScriptCloudflareWebAnalytics, useScriptFathomAnalytics, useScriptMatomoAnalytics, useScriptGoogleTagManager, useScriptG...
                                                                                                                                                                                                                                                                       ^
2: export { isVue2, isVue3 } from 'vue-demi';
3: export { defineNuxtLink } from '#app/components/nuxt-link';


  file: virtual:nuxt:.nuxt/imports.mjs:1:260

  1: export { useConsentScriptTrigger, useAnalyticsPageEvent, useElementScriptTrigger, useScript, useScriptGoogleAnalytics, useScriptPlausibleAnalytics, useScriptCloudflareWebAnalytics, useScriptFathomAnalytics, useScriptMatomoAnalytics, useScriptGoogleTagManager, useScriptG...
  ^
  2: export { isVue2, isVue3 } from 'vue-demi';
  3: export { defineNuxtLink } from '#app/components/nuxt-link';

  at getRollupError (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/parseAst.js:396:41)
  at error (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/parseAst.js:392:42)
  at Module.error (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:13858:16)
  at Module.getVariableForExportName (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:14022:29)
  at Module.includeAllExports (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:14098:37)
  at Module.includeDynamicImport (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:14579:28)
  at ImportExpression.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:10680:32)
  at ArrowFunctionExpression.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:8057:19)
  at ArrowFunctionExpression.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:8141:15)
  at ReturnValueScope.includeCallArguments (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:7582:26)
  at ArrowFunctionExpression.includeCallArguments (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:8061:20)
  at LocalVariable.includeCallArguments (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:6867:23)
  at Identifier.includeCallArguments (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:7007:23)
  at CallExpression.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:9184:21)
  at AwaitExpression.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:8524:23)
  at VariableDeclarator.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:10539:15)
  at VariableDeclaration.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:12214:28)
  at BlockStatement.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:7851:26)
  at FunctionExpression.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:8057:19)
  at FunctionExpression.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:8474:15)
  at Property.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:4822:23)
  at ObjectExpression.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:4818:28)
  at VariableDeclarator.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:10539:15)
  at VariableDeclaration.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:12214:28)
  at BlockStatement.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:7851:26)
  at FunctionDeclaration.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:8057:19)
  at FunctionDeclaration.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:8474:15)
  at ExportNamedDeclaration.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:4822:23)
  at Program.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:11372:22)
  at Module.include (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:14078:22)
  at Graph.includeStatements (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:20024:36)
  at Graph.build (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:19969:14)
  at async /home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:20611:13
  at async catchUnfinishedHookActions (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:20119:16)
  at async rollupInternal (/home/manu/Sites/unpress/node_modules/.pnpm/rollup@4.18.0/node_modules/rollup/dist/es/shared/node-entry.js:20606:5)
  at async Module.build (/home/manu/Sites/unpress/node_modules/.pnpm/vite@5.2.13_@types+node@20.14.2_terser@5.31.1/node_modules/vite/dist/node/chunks/dep-DEPSZ3SS.js:68734:18)
  at async buildClient (/home/manu/Sites/unpress/node_modules/.pnpm/@nuxt+vite-builder@3.12.1_@types+node@20.14.2_eslint@9.4.0_optionator@0.9.4_rollup@4.18.0_ter_xqi4of5p4nqos6vpjpslylp2n4/node_modules/@nuxt/vite-builder/dist/shared/vite-builder.ByBwRidQ.mjs:738:5)
  at async bundle (/home/manu/Sites/unpress/node_modules/.pnpm/@nuxt+vite-builder@3.12.1_@types+node@20.14.2_eslint@9.4.0_optionator@0.9.4_rollup@4.18.0_ter_xqi4of5p4nqos6vpjpslylp2n4/node_modules/@nuxt/vite-builder/dist/shared/vite-builder.ByBwRidQ.mjs:1661:3)
  at async bundle (/home/manu/Sites/unpress/node_modules/.pnpm/nuxt@3.12.1_@opentelemetry+api@1.9.0_@parcel+watcher@2.4.1_@types+node@20.14.2_@unocss+reset@_yxjhw5zvsqxaxtrtsod25wbmiu/node_modules/nuxt/dist/index.mjs:5391:5)
  at async Promise.all (index 1)
  at async build (/home/manu/Sites/unpress/node_modules/.pnpm/nuxt@3.12.1_@opentelemetry+api@1.9.0_@parcel+watcher@2.4.1_@types+node@20.14.2_@unocss+reset@_yxjhw5zvsqxaxtrtsod25wbmiu/node_modules/nuxt/dist/index.mjs:5266:5)
  at async Object.run (/home/manu/Sites/unpress/node_modules/.pnpm/nuxi@3.12.0/node_modules/nuxi/dist/chunks/build.mjs:94:5)
  at async runCommand$1 (/home/manu/Sites/unpress/node_modules/.pnpm/nuxi@3.12.0/node_modules/nuxi/dist/shared/nuxi.6aad497e.mjs:1648:16)
  at async runCommand$1 (/home/manu/Sites/unpress/node_modules/.pnpm/nuxi@3.12.0/node_modules/nuxi/dist/shared/nuxi.6aad497e.mjs:1639:11)
  at async runMain$1 (/home/manu/Sites/unpress/node_modules/.pnpm/nuxi@3.12.0/node_modules/nuxi/dist/shared/nuxi.6aad497e.mjs:1777:7)

/home/manu/Sites/unpress/packages/cms:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @unpress/cms@0.1.7-beta.1 build:standalone: `nuxi build`
Exit status 1
  ```
</details>

**Comment by huntz20** [comment]
hi @ManUtopiK i don't know it will help you or no but here's how i tackle this issue 

- remove node_modules, .nuxt, .lock file
- reinstall node_module
- build nuxt again
- then just uncomment all firebase setup run it and uncomment again


i know this is sound like magic but in my case it this issue is solve but i don't know will the issue will be persist in future or not

**Comment by danielroe** [comment]
this has been resolved with v3.12.2.
